frb_codegen/src/parser/mod.rs

```diff
@@ -16,9 +18,73 @@ use crate::method_utils::FunctionName;
 use crate::parser::ty::TypeParser;
 use crate::source_graph::Crate;
 
+use self::ty::convert_ident_str;
+
 const STREAM_SINK_IDENT: &str = "StreamSink";
 const RESULT_IDENT: &str = "Result";
 
+pub(crate) fn get_toposort(src: HashMap<String, Type>) -> HashMap<String, Type> {
+    // Some types that cannot be Handled.
+    // Filter something like `BareFn( TypeBareFn...`
+    // Filter something like `Ptr( TypePtr { star_token: Star,`
+    let mut ret: HashMap<String, Type> = src
+        .iter()
+        .filter_map(|(k, v)| match convert_ident_str(v) {
+            Some(_) => None,
+            None => Some((k.to_owned(), v.to_owned())),
+        })
+        .collect();
+
+    let string_src = src
+        .iter()
+        // Filter some types that cannot be Handled.
+        .filter_map(|(k, v)| convert_ident_str(v).map(|v| (k.to_owned(), v)));
+
+    let mut ts = TopologicalSort::<String>::new();
+
+    string_src.for_each(|(k, v)| {
+        // k and v switch orders here
+        ts.add_dependency(v, k);
+    });
+
+    // remove base type, like i32 in `type Id = i32`.
+    // You might worry about the type, which cannot be handled and isn't in ts.
+    // Case:
+    // ```
+    // type Id = unsafe{};
+    // type UserId = Id;
+    // ```
+    // 1. pop_base: ret = [{Id, unsafe}]
+    // 2. init_condition: pop Id, and then handle UserId.
+    //    src.get("UserId") => Id,
```

Comment:
sorry but I cannot quite understand the comments. would be great to rewrite a bit to make it more clear for (future) readers

Reply:
// remove base type, like i32 in `type Id = i32`.
    // You might worry about the type, which cannot be handled and isn't in ts.
    // Case:
    // ```
    // type UnsafeAlias = unsafe{};
    // type NestAlias = UnsafeAlias;
    // ```
    // 1. pop_base: ret = [{UnsafeAlias, unsafe}]
    // 2. init_condition: pop UnsafeAlias, and then handle NestAlias.
    //    src.get("NestAlias") => UnsafeAlias,

It is need to modify later